internal/jptrace/aggregator.go

```diff
@@ -0,0 +1,43 @@
+package jptrace
+
+import (
+	"iter"
+	"runtime/trace"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func AggregateTraces(tracesSeq iter.Seq2[[]ptrace.Traces, error]) iter.Seq2[ptrace.Traces, error] {
+	return func(yield func(trace ptrace.Traces, err error) bool) {
+		trace.IsEnabled()
+		currentTrace := ptrace.NewTraces()
+		currentTraceID := pcommon.NewTraceIDEmpty()
+
+		tracesSeq(func(traces []ptrace.Traces, err error) bool {
+			if err != nil {
+				yield(ptrace.NewTraces(), err)
+				return false
+			}
+			for _, trace := range traces {
+				resources := trace.ResourceSpans()
+				// TODO: add non-empty requirement to contract
+				traceID := resources.At(0).ScopeSpans().At(0).Spans().At(0).TraceID()
+				if currentTraceID == traceID {
+					// TODO: merge traces
```

Comment:
@yurishkuro can the merge function simply be the following? 
```go
func mergeTraces(src, dest ptrace.Traces) {
	src.ResourceSpans().CopyTo(dest.ResourceSpans())
}
```

Reply:
wouldn't that override the spans? We want to append them

Reply:
@yurishkuro yes sorry - i meant something more like this
```go
func mergeTraces(src, dest ptrace.Traces) {
	resources := src.ResourceSpans()
	for i := 0; i < resources.Len(); i++ {
		resource := resources.At(i)
		resource.CopyTo(dest.ResourceSpans().AppendEmpty())
	}
}
```

Reply:
yep